### PR TITLE
OCPBUGS-17785: Document SR-IOV/FlowCollector config

### PR DESCRIPTION
**Pull request by skrthomas**
Version(s):

4.11+
Issue:

eng ticket: https://issues.redhat.com/browse/OCPBUGS-17785 
docs ticket: https://issues.redhat.com/browse/OCPBUGS-18891
Link to docs preview:

https://64448--docspreview.netlify.app/openshift-enterprise/latest/networking/network_observability/configuring-operator#network-observability-SR-IOV-config_network_observability
QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/network-observability-SRIOV-configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+//
+// network_observability/configuring-operator.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-SR-IOV-config_{context}"]
+= Configure monitoring of SR-IOV interface traffic
+In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors the host network namespaces. When a pod with virtual functions (VF) interface is created, a new network namespace is created and the VF interface is migrated from the host network namespace to the pod network namespace.
```
Comment (msherif1234) [inline modules/network-observability-SRIOV-configuration.adoc:8]: the need for `privileged` to be able to detect none host netns, today ebpf agent by default watch host netns 

also the VF migrated under certain configuration in sriov policy where we have to specify `IPAM` without no migration will take place, @memodi can share more configs here as needed

Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc:8]: Discussed with @msherif1234 on Slack and the following updates will be made based on this comment:
- Clarify that egardless to the privileged setting host netns is always visible AND to see other netns we have to set privileged to true
-  make it clear VF interfaces won't always migrate from host netns to pod netns; it only migrates under certain configurations. Add for example if SRIOV policy uses IPAM.

Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc]: Addressed. I added the `SRIOVNetwork` CR IPAM config to the `Prerequisites`. Is that OK @msherif1234 ?

File: modules/network-observability-SRIOV-configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// network_observability/configuring-operator.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-SR-IOV-config_{context}"]
+= Configuring the monitoring of SR-IOV interface traffic
+In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual functions (VF) interface is created, a new network namespace is created. The VF interface is migrated from the host network namespace to the pod network namespace.
+
```
Comment (msherif1234) [inline modules/network-observability-SRIOV-configuration.adoc:9]: that isn't always the case though when using special sriov policy config that can happen also cc'd @bn222 for his comments

Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc:9]: Ok...I updated this to say "With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace." Does that help? Or could be clearer?

File: modules/network-observability-SRIOV-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// network_observability/configuring-operator.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-SR-IOV-config_{context}"]
+= Configuring the monitoring of SR-IOV interface traffic
+In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual functions (VF) interface is created, a new network namespace is created. With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace.
+
```
Comment (memodi) [inline modules/network-observability-SRIOV-configuration.adoc:9]: > the VF interface is migrated from the host network namespace to the pod network namespace.

is this technical detail required in docs?

**Review by memodi — COMMENTED** [review]
Should we mention something out UI where flows from these network namespaces are not enriched (we have RFE for it) ? They can still filter in UI by interface but they won't see related ns/workloads/pods . cc @msherif1234

File: modules/network-observability-SRIOV-configuration.adoc (hunk 3 above)
Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc:9]: @memodi I think @msherif1234 was wanting to make really clear that the 'SRIOVNetwork` spec is crucial to the migration. I'll let him weigh in and see if maybe we should tone down the technical details here.

Comment (memodi) [reply modules/network-observability-SRIOV-configuration.adoc:9]: Got it, I understand it better now, we can that info ON. Resolving this.

**Comment by msherif1234** [comment]
> Should we mention something out UI where flows from these network namespaces are not enriched (we have RFE for it) ? They can still filter in UI by interface but they won't see related ns/workloads/pods . cc @msherif1234
I am not sure if this has anything to do with SRIOV or its because how the test was running 
I think we don't hvae to and if it came out then we already have 
https://issues.redhat.com/browse/NETOBSERV-1289

**Comment by bergerhoffer** [comment]
@skrthomas I'll find it now, but can you add the preview link to the description when you get a chance?

File: modules/network-observability-SRIOV-configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+.Prerequisite
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:10]: Should be plural

```suggestion
.Prerequisites
```

File: modules/network-observability-SRIOV-configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// network_observability/configuring-operator.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-SR-IOV-config_{context}"]
+= Configuring the monitoring of SR-IOV interface traffic
+In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual functions (VF) interface is created, a new network namespace is created. With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace.
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:8]: * Should this have "a" in front"?
* I know people call these different things, but I've never seen "specification value" used. Maybe just say "field"?
* I'm not sure if starting the second sentence with "Then" makes sense, because you didn't tell them an action to take, you just made a statement saying that it has to be true. Maybe if you change the first sentence to say "you must set the .... to `true`." it would make better sense? And it would be more active, so that might be the way to go, if it makes sense. If it doesn't make sense (because maybe it's not this person that set it?) then maybe reconsider replacing "Then" with a different phrase.
* Should virtual function be singular? I see it both ways in the repo

```suggestion
In order to collect traffic from a cluster with a Single Root I/O Virtualization (SR-IOV) device, you must set the `FlowCollector` `spec.agent.ebpf.privileged` field value to `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual function (VF) interface is created, a new network namespace is created. With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace.
```

File: modules/network-observability-SRIOV-configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+.Prerequisite
+* Access to an {product-title} cluster with SR-IOV device.
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:11]: Same question on if there should be an "a" in front of "SR-IOV device"?

File: modules/network-observability-SRIOV-configuration.adoc [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+// network_observability/configuring-operator.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-SR-IOV-config_{context}"]
+= Configuring the monitoring of SR-IOV interface traffic
+In order to collect traffic from a cluster with Single Root I/O Virtualization (SR-IOV) device, the `FlowCollector` `spec.agent.ebpf.privileged` specification value must be `true`. Then, the eBPF agent monitors other network namespaces in addition to the host network namespaces, which are monitored by default. When a pod with a virtual functions (VF) interface is created, a new network namespace is created. With `SRIOVNetwork` policy `IPAM` configurations specified, the VF interface is migrated from the host network namespace to the pod network namespace.
+
+.Prerequisite
+* Access to an {product-title} cluster with SR-IOV device. 
+* The `SRIOVNetwork` custom resource (CR) specifies `IPAM` configured with an IP address from the range in which interface has the IP or from other plugins.
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:12]: I'm unsure what this is saying, is there a "the" missing in front of "interface"? Also the "in which" might be throwing me off. And maybe I don't understand what the "or from other plugins" refers to. Like what's the alternative to other plugins?

I know this isn't my area of expertise so I can't give a suggestion, but I think it's hard to follow what you mean here, so maybe worth another look

File: modules/network-observability-SRIOV-configuration.adoc [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+.Procedure
+. In the web console, navigate to *Operators* -> *Installed Operators*.
+. Under the *Provided APIs* heading for the *NetObserv Operator*, select *Flow Collector*. 
+. Select *cluster* then select the *YAML* tab.
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:17]:
```suggestion
. Select *cluster* and then select the *YAML* tab.
```

File: modules/network-observability-SRIOV-configuration.adoc [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  name: cluster
+spec:
+  namespace: netobserv
+  deploymentModel: DIRECT
+  agent:
+    type: EBPF
+    ebpf:        
+      privileged: true             <1> 
+----
+<1> Note that the `spec.agent.ebpf.privileged` specification value must be `true` for packet drop tracking to be enabled.
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:36]: * I don't think the "Note that" is necessary
* Same comment about specification value

```suggestion
<1> The `spec.agent.ebpf.privileged` field value must be `true` to enable packet drop tracking.
```

File: modules/network-observability-SRIOV-configuration.adoc [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+kind: FlowCollector
+metadata:
+  name: cluster
+spec:
+  namespace: netobserv
+  deploymentModel: DIRECT
+  agent:
+    type: EBPF
+    ebpf:        
+      privileged: true             <1>
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:34]: Should the <1> be just a single space after "true" here? I know that I sometimes put mine all right aligned when I have multiple, but this is just a single one, and it isn't right aligned past the longest line (line 24)

File: modules/network-observability-SRIOV-configuration.adoc [hunk 11]
```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// network_observability/configuring-operator.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-SR-IOV-config_{context}"]
+= Configuring the monitoring of SR-IOV interface traffic
```
Comment (bergerhoffer) [inline modules/network-observability-SRIOV-configuration.adoc:7]: Title is fine as is, but adding an alternative suggestion of "Configuring monitoring for SR-IOV interface traffic"

**Review by bergerhoffer — COMMENTED** [review]
Just a few suggestions!

File: modules/network-observability-SRIOV-configuration.adoc (hunk 5 above)
Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc:8]: - I do think I'm missing an article here, good catch. 
- People really do call these different things. I think field is good. I think someone has recommended that I specifically say `value` in another peer review but I don't know for sure. 
- I think changing to "you must set the ... to `true`" is a good suggestion, and I can definitely see how it makes more sense with the "Then".
- I think in this case it is singular.

File: modules/network-observability-SRIOV-configuration.adoc (hunk 7 above)
Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc:12]: Yeah this could definitely be clearer, you're right. This was written in a crunch, and it definitely shows. I think this may be more like it:
The `SRIOVNetwork` custom resource (CR) `spec.ipam` configuration must be set with an IP address from the range that the interface lists or from other plugins.

File: modules/network-observability-SRIOV-configuration.adoc (hunk 10 above)
Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc:34]: Yeah I clearly got really happy with my space bar 😆 Could definitely have less spacing.

File: modules/network-observability-SRIOV-configuration.adoc (hunk 11 above)
Comment (skrthomas) [reply modules/network-observability-SRIOV-configuration.adoc]: Your suggestion is much cleaner, thank you!
